packages/app-store/closecom/pages/setup/index.tsx

```diff
@@ -63,13 +65,17 @@ export default function CloseComSetup() {
               <Form
                 form={form}
                 handleSubmit={async (values) => {
-                  const res = await fetch("/api/integrations/closecom/add", {
-                    method: "POST",
-                    body: JSON.stringify(values),
-                    headers: {
-                      "Content-Type": "application/json",
-                    },
-                  });
+                  const { returnTo } = query;
+                  const res = await fetch(
+                    `/api/integrations/closecom/add${returnTo ? `?returnTo=${returnTo}` : ""}`,
+                    {
```

Comment:
NIT:

```suggestion
                  const url = new URL('/api/integrations/closecom/add');
                  if (returnTo) url.searchParams.append('returnTo', returnTo);
                  const res = await fetch(url.href, {
```

Reply:
opened a new issue for the NIT: https://github.com/calcom/cal.com/issues/14804